### ISSUE TITLE
ODataModelBinderConverter is broken in .Net 6.0

### ISSUE THREAD

**Issue by dxrdxr** [OData/AspNetCoreOData#352]
Build AspNetCoreOData with a target of net6.0.

Run tests.

Failures will occur in `ODataModelBinderConverter`.

Problem code:
```
private static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethods()
    .Single(m => m.Name == "TryParse" && m.GetParameters().Length == 2);
```

In .Net 6.0, `Enum` has two methods named `TryParse` that have exactly 2 parameters (see [here](https://github.com/dotnet/runtime/issues/1916)):

```
TryParse<TEnum>(String, TEnum)
TryParse<TEnum>(ReadOnlySpan<Char>, Boolean, TEnum)
```
The `ReadOnlySpan` is new and causes `EnumTryParseMethod ` to fail as there is not a `Single` method.

Suggested fix (minimally tested):

```
private static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethods()
    .Single(m => m.Name == "TryParse" && m.GetParameters().Length == 2 && m.GetParameters()[0].ParameterType == typeof(string));
```

**Comment by xuzhg** [comment]
@dxrdxr It seems it's same as the fix at: https://github.com/OData/AspNetCoreOData/blob/master/src/Microsoft.AspNetCore.OData/Query/Expressions/ExpressionBinderBase.cs#L43-L51

**Comment by xuzhg** [comment]
@dxrdxr I did the same fix at e0c5245bb6585063649cb50c02099e619d9d4153.
